rts/lra/coordinator/src/main/java/io/narayana/lra/coordinator/api/Coordinator.java

```diff
@@ -134,20 +105,6 @@ public class Coordinator extends Application {
     private final LRAService lraService;
     private final RecoveryCoordinator recoveryCoordinator;
 
-    @Override
-    public Set<Class<?>> getClasses() {
-        HashSet<Class<?>> classes = new HashSet<>();
-        classes.add(Coordinator.class);
-        return classes;
-    }
-
-    @Override
-    public Set<Object> getSingletons() {
```

Comment:
I notice that this method is completely going. I don't know why it was present before, do you know?

Reply:
I am thinking we might want to keep this somewhere - https://docs.oracle.com/javaee/7/api/javax/ws/rs/core/Application.html#getSingletons-- - unless we no longer need CoordinatorContainerFilter to be a singleton?

Reply:
(I am asking elsewhere in this PR if the empty beans.xml is somehow related to this)

Reply:
It is singleton by default - https://quarkus.io/guides/resteasy#lifecycle-of-resources

Reply:
For clarity, are you saying that https://github.com/jbosstm/narayana/blob/master/rts/lra/coordinator/src/main/java/io/narayana/lra/coordinator/api/CoordinatorContainerFilter.java is a Singleton by default? And is it the same for other containers (like WildFly)?

Reply:
It is a singleton by default as stated at the documentation I posted above. I'm sorry Tom, but to answer your second question I would need to do the research which point I don't see as it is irrelevant to this PR and Quarkus.

Reply:
In particular I would like to know if the singletons thing is something in Quarkus or defined by a spec

Reply:
Yes it does seem like there are a lot more changes/deletions than the issue report would suggest. If we need to back out your OpenAPI.Swagger changes then we will end up removing a lot more than just the annotations. Ideally we would want these extra "clean up"/"whatever the rationale is" changes to go into a separate commit. In earlier versions of the JAX-RS spec this was the standard way to declare resources (and it's still supported).

Will you update the issue indicating why you need the extra changes, over and above the changes to the OpenAPI annotations.

Reply:
Regarding CoordinatorContainerFilter.java (https://github.com/jbosstm/narayana/blob/master/rts/lra/coordinator/src/main/java/io/narayana/lra/coordinator/api/CoordinatorContainerFilter.java#L53) I think it seems the default for Providers per Application in EE 10 is a single instance (https://jakarta.ee/specifications/restful-ws/3.1/jakarta-restful-ws-spec-3.1.html#lifecycle_and_environment)

Reply:
Regarding the original question in this thread, it is resolved from my point of view and limited understanding of the it.